static/js/types.ts

```diff
@@ -1,3 +1,31 @@
+// TODO/typescript: Move this to message_store
+export type MatchedMessage = {
+    match_content: string;
+    match_subject: string;
```

Comment:
I think `match_content` and `match_subject` will always be potentially `undefined`, so we might as well declare them optional here and remove the usage of `Partial`.

Reply:
Fixed.